tests/transformers/test_transformers_model_export.py

```diff
@@ -3865,3 +3865,40 @@ def test_basic_model_with_accelerate_homogeneous_mapping_works(tmp_path):
     text = "Apples are delicious"
 
     assert loaded(text) == pipeline(text)
+
+
+def test_qa_model_model_size_bytes(small_qa_pipeline, tmp_path):
+    def _calculate_expected_size(path_or_dir):
+        # this helper function does not consider subdirectories
+        expected_size = 0
+        if os.path.isdir(path_or_dir):
+            for path in os.listdir(path_or_dir):
+                path = os.path.join(path_or_dir, path)
+                if not os.path.isfile(path):
+                    continue
+                with open(path, "rb") as fp:
+                    expected_size += len(fp.read())
+        elif os.path.isfile(path_or_dir):
+            with open(path_or_dir, "rb") as fp:
+                expected_size = len(fp.read())
+        return expected_size
+
+    mlflow.transformers.save_model(
+        transformers_model=small_qa_pipeline,
+        path=tmp_path,
+    )
+
+    # expected size only counts for files saved before the MLmodel file is saved
+    model_dir = tmp_path.joinpath("model")
+    tokenizer_dir = tmp_path.joinpath("components").joinpath("tokenizer")
+    expected_size = 0
+    for folder in [model_dir, tokenizer_dir]:
+        folder = str(folder)
+        expected_size += _calculate_expected_size(folder)
+    other_files = ["model_card.md", "model_card_data.yaml"]
+    for file in other_files:
+        path = str(tmp_path.joinpath(file))
+        expected_size += _calculate_expected_size(path)
+
+    mlmodel = yaml.safe_load(tmp_path.joinpath("MLmodel").read_bytes())
+    assert mlmodel["model_size_bytes"] == expected_size
```

Comment:
/ai rewrite this by leveraging pathlib.Path methods. Note tmp_path is a pathlib.Path object.